### DISCUSSION TITLE
Get help about implicit methods in Sequelize

### DISCUSSION THREAD

**Discussion by GreenFlag31** [sequelize/sequelize#17283] [Help & Questions]
Hi.

I just started Sequelize on week ago and would like some help/documentation/references about the usage of Reflection (or implicit methods) that Sequelize create.

This example comes from the documentation:
```javascript
const amidala = await User.create({ username: 'p4dm3', points: 1000 });
const queen = await Profile.create({ name: 'Queen' });
await amidala.addProfile(queen, { through: { selfGranted: false } });
const result = await User.findOne({
  where: { username: 'p4dm3' },
  include: Profile,
});
console.log(result);
```

I would like some information about the `addProfile` method that is defined nowhere. Where can I find documentation about this? I use Typescript with NodeJS and I'm forced to cast to any, otherwise I got a 'method not exist' compilation error from ts.
`await (amidala as any).addProfile(queen, { through: { selfGranted: false } });`

Could you tell me more about this (documentation about Reflection) and how to properly cast to avoid any?

Thanks.

**Comment by WikiRik** [comment]
If you're new to Sequelize and don't have a project that will go to production soon, it might be good to check out the v7-alphas. They work a lot better with TypeScript. See for example this documentation; https://sequelize.org/docs/v7/associations/belongs-to-many/#association-methods
Do note that v7 is installed from `@sequelize/core` instead of `sequelize` and the general setup is also different. Check out the Getting Started guide for v7 here; https://sequelize.org/docs/v7/getting-started/

**Reply by GreenFlag31** [reply]
Thanks, will take a look.

Meanwhile, I found a way [in the sequelize documentation about Typescript](https://sequelize.org/docs/v6/other-topics/typescript/#usage), but things have to be manually created. In my opinion, a part in the documentation should be added regarding auto generated methods. I have searched nearly the whole doc and haven't found satisfying info about it.

```javascript
  declare getProjects: HasManyGetAssociationsMixin<Project>;
  declare addProject: HasManyAddAssociationMixin<Project, number>;
  declare addProjects: HasManyAddAssociationsMixin<Project, number>;
  declare setProjects: HasManySetAssociationsMixin<Project, number>;
  declare removeProject: HasManyRemoveAssociationMixin<Project, number>;
  ...
```
